### PR TITLE
fix: route AKS log collection output to log file

### PR DESCRIPTION
**Pull request by cameronmeissner**
**What type of PR is this?**

**What this PR does / why we need it**:

Right now the response RP gets back from bootstrapping nodes when CSE fails is polluted with the output of the log collection script - specifically the `tar` calls where all the files included in the tarball are enumerated.

For example:

```
"errorDetails": VM has reported a failure when processing extension 'vmssCSE' (publisher 'Microsoft.Azure.Extensions' and type 'CustomScript'). Error message: 'Enable failed: failed to execute command: command terminated with exit status=51
[stdout]
ing: etc/cni/net.d/10-azure.conflist (deflated 52%)
  adding: etc/containerd/config.toml (deflated 66%)
  adding: etc/containerd/kubenet_template.conf (deflated 49%)
  adding: etc/default/kubelet (deflated 55%)
  adding: var/log/azure-cnimonitor.log (stored 0%)
  adding: var/log/azure-vnet-ipam.log (stored 0%)
  adding: var/log/azure-vnet-telemetry.log (stored 0%)
  adding: var/log/azure-vnet.log (stored 0%)
  adding: var/lib/waagent/provisioned (stored 0%)
  adding: etc/fstab (deflated 22%)
  adding: etc/ssh/sshd_config (deflated 54%)
  adding: boot/grub/grub.cfg (deflated 76%)
  adding: etc/lsb-release (deflated 24%)
  adding: etc/os-release (deflated 38%)
  adding: etc/hostname (deflated 3%)
  adding: etc/apt/sources.list (deflated 73%)
  adding: etc/apt/sources.list.d/microsoft-prod-testing.list (deflated 7%)
  adding: etc/apt/sources.list.d/microsoft-prod.list (deflated 5%)
  adding: etc/netplan/50-cloud-init.yaml (deflated 44%)
  adding: etc/nsswitch.conf (deflated 50%)
  adding: etc/resolv.conf (deflated 46%)
  adding: run/systemd/resolve/stub-resolv.conf (deflated 48%)
  adding: etc/ufw/ufw.conf (deflated 31%)
  adding: etc/waagent.conf (deflated 54%)
  adding: var/lib/hyperv/.kvp_pool_0 (stored 0%)
  adding: var/lib/hyperv/.kvp_pool_1 (deflated 92%)
  adding: var/lib/hyperv/.kvp_pool_2 (stored 0%)
  adding: var/lib/hyperv/.kvp_pool_3 (deflated 99%)
  adding: var/lib/hyperv/.kvp_pool_4 (stored 0%)
  adding: var/log/azure/custom-script/handler.log (deflated 84%)
  adding: var/lib/waagent/ovf-env.xml (deflated 35%)
  adding: var/lib/waagent/Microsoft.AKS.Compute.AKS.Linux.Billing-1.0.0/status/1.status (deflated 50%)
  adding: var/lib/waagent/Microsoft.Azure.Extensions.CustomScript-2.1.10/status/1.status (deflated 31%)
  adding: var/lib/waagent/Microsoft.AKS.Compute.AKS.Linux.Billing-1.0.0/config/1.settings (deflated 37%)
  adding: var/lib/waagent/Microsoft.Azure.Extensions.CustomScript-2.1.10/config/1.settings (deflated 24%)
  adding: var/lib/waagent/Microsoft.AKS.Compute.AKS.Linux.Billing-1.0.0/config/HandlerState (stored 0%)
  adding: var/lib/waagent/Microsoft.Azure.Extensions.CustomScript-2.1.10/config/HandlerState (stored 0%)
  adding: var/lib/waagent/Microsoft.AKS.Compute.AKS.Linux.Billing-1.0.0/config/HandlerStatus (deflated 24%)
  adding: var/lib/waagent/Microsoft.Azure.Extensions.CustomScript-2.1.10/config/HandlerStatus (deflated 24%)
  adding: var/lib/waagent/SharedConfig.xml (deflated 51%)
  adding: var/log/cloud-init-output.log (deflated 72%)
  adding: var/log/cloud-init.log (deflated 87%)
  adding: var/log/azure/Microsoft.AKS.Compute.AKS.Linux.Billing/CommandExecution.log (deflated 75%)
  adding: var/log/azure/Microsoft.Azure.Extensions.CustomScript/CommandExecution.log (deflated 82%)
  adding: var/log/azure/aks/cloud-config.txt (deflated 33%)
  adding: var/log/azure/aks/cluster-provision-cse-output.log (deflated 70%)
  adding: var/log/azure/aks/cluster-provision.log (deflated 80%)
  adding: var/log/azure/aks/components.json (deflated 86%)
  adding: var/log/azure/aks/kube-proxy-images.json (deflated 66%)
  adding: var/log/azure/aks/manifest.json (deflated 75%)
  adding: var/log/azure/aks/provision.json (deflated 65%)
  adding: var/log/azure/aks/vhd-install.complete (deflated 82%)
  adding: var/log/azure/Microsoft.Azure.Extensions.CustomScript/events/1715796795096.json (deflated 56%)
  adding: var/log/syslog (deflated 92%)
  adding: var/log/syslog.1 (deflated 91%)
  adding: var/log/messages (deflated 92%)
  adding: var/log/messages.1 (deflated 91%)
  adding: var/log/kern.log (deflated 94%)
  adding: var/log/kern.log.1 (deflated 93%)
  adding: var/log/dmesg (deflated 73%)
  adding: var/log/dmesg.0 (deflated 72%)
  adding: var/log/dmesg.1.gz (deflated 10%)
  adding: var/log/dmesg.2.gz (deflated 9%)
  adding: var/log/dmesg.3.gz (stored 0%)
  adding: var/log/dpkg.log (deflated 92%)
  adding: var/log/auth.log (deflated 63%)
  adding: var/log/auth.log.1 (deflated 89%)
Log bundle size: 1.6M	aks_logs.zip
Uploading log bundle: Successfully uploaded logs
Cleaning up /tmp/tmp.rYHGdHMGNc...
Log collection finished.

[stderr]
'. More information on troubleshooting is available at https://aka.ms/VMExtensionCSELinuxTroubleshoot. ,
"httpStatusCode": 200,
"durationInMilliseconds": 1328
```

 Before the introduction of the log collector, the snippet of CSE logs that RP would get back from bootstrapping nodes contained some helpful information, usually pertaining to exactly _where_ the corresponding exit code was being thrown.

This PR aims to restore the previously meaningful CSE responses.

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

**Comment by djsly** [comment]
@cameronmeissner  what is this trying to fix ?

**Comment by cameronmeissner** [comment]
> @cameronmeissner what is this trying to fix ?

updated the description

File: parts/linux/cloud-init/artifacts/cse_start.sh [hunk 1]
```diff
@@ -85,7 +85,7 @@ upload_logs() {
     # fall back to WALA collector
     if test -x /opt/azure/containers/aks-log-collector.sh; then
         # Call AKS Log Collector
-        /opt/azure/containers/aks-log-collector.sh
+        /opt/azure/containers/aks-log-collector.sh >/dev/null 2>&1
```
Comment (phealy) [inline parts/linux/cloud-init/artifacts/cse_start.sh:88]: Dumping it to /dev/null makes it really hard to troubleshoot any problems later - would it be better to dump it to /var/log/azure/aks/cse-aks-log-collector.log or something similar?

Comment (djsly) [reply parts/linux/cloud-init/artifacts/cse_start.sh:88]: I"m in favour of keeping the traceability.  We might even want to grep in the future of common error in that log file

**Comment by cameronmeissner** [comment]
from agentbaker e2e's

VM has reported a failure when processing extension 'vmssCSE' (publisher 'Microsoft.Azure.Extensions' and type 'CustomScript'). Error message: 'Enable failed: failed to execute command: command terminated with exit status=124 [stdout] { "ExitCode": "124", "Output": "='Server:\t\t168.63.129.16\nAddress:\t168.63.129.16#53\n\nNon-authoritative answer:\nName:\tagentbaker-e2e-test-cluster-vfozt-ua7id57w.hcp.eastus.azmk8s.io\nAddress: 40.88.231.78\n\nExecuted nslookup -timeout=15 -retry=0 agentbaker-e2e-test-cluster-vfozt-ua7id57w.hcp.eastus.azmk8s.io for 1 seconds'\n+ STS=0\n+ [[ 0 != 0 ]]\n+ logs_to_events AKS.CSE.apiserverNC 'retrycmd_if_failure 50 1 10 nc -vz agentbaker-e2e-test-cluster-vfozt-ua7id57w.hcp.eastus.azmk8s.io 443'\n+ local task=AKS.CSE.apiserverNC\n+ shift\n++ date +%s%3N\n+ local eventsFileName=1716583047614\n++ date '+%F %T.%3N'\n+ local 'startTime=2024-05-24 20:37:27.616'\n+ retrycmd_if_failure 50 1 10 nc -vz agentbaker-e2e-test-cluster-vfozt-ua7id57w.hcp.eastus.azmk8s.io 443\n+ retries=50\n+ wait_sleep=1\n+ timeout=10\n+ shift\n+ shift\n+ shift\n++ seq 1 50\n+ for i in $(seq 1 $retries)\n+ timeout 10 nc -vz agentbaker-e2e-test-cluster-vfozt-ua7id57w.hcp.eastus.azmk8s.io 443\nConnection to agentbaker-e2e-test-cluster-vfozt-ua7id57w.hcp.eastus.azmk8s.io (40.88.231.78) 443 port [tcp/https] succeeded!\n+ break\n+ echo Executed '\"nc' -vz agentbaker-e2e-test-cluster-vfozt-ua7id57w.hcp.eastus.azmk8s.io '443\"' 1 times\nExecuted \"nc -vz agentbaker-e2e-test-cluster-vfozt-ua7id57w.hcp.eastus.azmk8s.io 443\" 1 times\n+ ret=0\n++ date '+%F %T.%3N'\n+ local 'endTime=2024-05-24 20:37:27.637'\n++ jq -n --arg Timestamp '2024-05-24 20:37:27.616' --arg OperationId '2024-05-24 20:37:27.637' --arg Version 1.23 --arg TaskName AKS.CSE.apiserverNC --arg EventLevel Informational --arg Message 'Completed: retrycmd_if_failure 50 1 10 nc -vz agentbaker-e2e-test-cluster-vfozt-ua7id57w.hcp.eastus.azmk8s.io 443' --arg EventPid 0 --arg EventTid 0 '{Timestamp: $Timestamp, OperationId: $OperationId, Version: $Version, TaskName: $TaskName, EventLevel: $EventLevel, Message: $Message, EventPid: $EventPid, EventTid: $EventTid}'\n+ json_string='{\n \"Timestamp\": \"2024-05-24 20:37:27.616\",\n \"OperationId\": \"2024-05-24 20:37:27.637\",\n \"Version\": \"1.23\",\n \"TaskName\": \"AKS.CSE.apiserverNC\",\n \"EventLevel\": \"Informational\",\n \"Message\": \"Completed: retrycmd_if_failure 50 1 10 nc -vz agentbaker-e2e-test-cluster-vfozt-ua7id57w.hcp.eastus.azmk8s.io 443\",\n \"EventPid\": \"0\",\n \"EventTid\": \"0\"\n}'\n+ echo '{' '\"Timestamp\":' '\"2024-05-24' '20:37:27.616\",' '\"OperationId\":' '\"2024-05-24' '20:37:27.637\",' '\"Version\":' '\"1.23\",' '\"TaskName\":' '\"AKS.CSE.apiserverNC\",' '\"EventLevel\":' '\"Informational\",' '\"Message\":' '\"Completed:' retrycmd_if_failure 50 1 10 nc -vz agentbaker-e2e-test-cluster-vfozt-ua7id57w.hcp.eastus.azmk8s.io '443\",' '\"EventPid\":' '\"0\",' '\"EventTid\":' '\"0\"' '}'\n+ '[' 0 '!=' 0 ']'\n+ [[ ubuntu != \\m\\a\\r\\i\\n\\e\\r ]]\n++ date\n+ echo 'Recreating man-db auto-update flag file and kicking off man-db update process at Fri May 24 20:37:27 UTC 2024'\nRecreating man-db auto-update flag file and kicking off man-db update process at Fri May 24 20:37:27 UTC 2024\n+ createManDbAutoUpdateFlagFile\n+ touch /var/lib/man-db/auto-update\n+ echo 'I am sad...'\nI am sad...\n+ exit 124\n+ /usr/bin/mandb", "Error": "", "ExecDuration": "24", "KernelStartTime": "Fri 2024-05-24 20:36:25 UTC", "CloudInitLocalStartTime": "Fri 2024-05-24 20:36:27 UTC", "CloudInitStartTime": "Fri 2024-05-24 20:36:30 UTC", "CloudFinalStartTime": "Fri 2024-05-24 20:37:02 UTC", "NetworkdStartTime": "Fri 2024-05-24 20:36:29 UTC", "CSEStartTime": "Fri May 24 20:37:03 UTC 2024", "GuestAgentStartTime": "Fri 2024-05-24 20:36:47 UTC", "SystemdSummary": "Startup finished in 507ms (firmware) + 4.033s (loader) + 1.251s (kernel) + 36.599s (userspace) = 42.391s \ngraphical.target reached after 35.806s in userspace", "BootDatapoints": { "KernelStartTime": "Fri 2024-05-24 20:36:25 UTC", "CSEStartTime": "Fri May 24 20:37:03 UTC 2024", "GuestAgentStartTime": "Fri 2024-05-24 20:36:47 UTC", "KubeletStartTime": "Fri 2024-05-24 20:37:25 UTC" } } [stderr] '. More information on troubleshooting is available at https://aka.ms/VMExtensionCSELinuxTroubleshoot.

**Review by phealy — APPROVED** [review]

**Review by djsly — APPROVED** [review]

**Pull request merged**